File: backend/src/main/java/bg/pcbudget/backend/models/Monitor.java
package bg.pcbudget.backend.models;

import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@Table(name = "monitors")
@Getter
@Setter
@NoArgsConstructor
public class Monitor extends Product {

  private Double panelSizeInch;
  private String aspectRatio;
  private String resolution;
  private Integer refreshRateHz;
  private Integer responseTimeMs;
  private String panelType;
  private Integer brightnessNits;

  @ManyToMany
  @JoinTable(
    name = "monitor_interfaces",
    joinColumns = @JoinColumn(name = "monitor_id"),
    inverseJoinColumns = @JoinColumn(name = "interface_id")
  )
  private List<ItemInterface> interfaces;
}

File: src/app/models/Monitor.ts
import { Product } from './Product';
import { Interface } from './Interface';

export interface Monitor extends Product {
  panelSizeInch:   number;
  aspectRatio:     string;
  resolution:      string;
  refreshRateHz:   number;
  responseTimeMs:  number;
  panelType:       string;
  brightnessNits:  number;
  interfaces:      Interface[];
}
